tests/functional_tests/p2p.py

```diff
@@ -183,6 +226,42 @@ def test_p2p_tx_propagation(self):
             assert len(res.tx_hashes) == 1
             assert res.tx_hashes[0] == txid
 
+        return txid
+
+    def test_p2p_block_propagation(self, daemon2_mempool_txid):
+        print('Testing P2P block propagation')
+        daemon2 = Daemon(idx = 2)
+        daemon3 = Daemon(idx = 3)
+
+        # check precondition: txid in daemon2 mempool
```

Comment:
Probably easier to follow if you just fold this test into the `tx_propagation` test at the end, as it's quite intimately related (the first part would test pooled tx propagation, and the second part tx in block propagation). For an example of the overlap, the "precondition" here vanishes, because it's already been checked at the end of the prior function.

Reply:
I disagree, they really are testing different things, but there's just a precondition that overlaps with the previous test case.

Reply:
If you want it to be wholly separate, then you could make it that way by explicitly adding a transaction to the pool in this test. Having tests depend on state configured by other tests makes failures all the more difficult to comprehend. I won't belabor the point because the functional test suite has so many structural flaws.